src/main/kotlin/app/batch/processor/DecryptionProcessor.kt

```diff
@@ -30,9 +33,7 @@ class DecryptionProcessor(private val cipherService: CipherService,
                             decryptedKey,
                             item.encryption.initializationVector,
                             item.dbObject)
-            val jsonObject = Gson().fromJson(decrypted, JsonObject::class.java)
-            jsonObject.addProperty("timestamp", item.hbaseTimestamp)
-            return DecryptedRecord(jsonObject, item.db, item.collection)
+            return skipBadDecryptedRecords(item, decrypted)
```

Comment:
Could we move the filtering of bad records to a separate processor - if it returns null when fields are missing batch will skip it - or stick with the exception based approach.

Reply:
Dan , i have delegated the code to Validator which throws a baddecrypteddata exception that is  configured to be skippable in job configuration